components/tensorflow-notebook-image/pvc-check.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+
+# Copyright 2016 The Kubeflow Authors All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# stored in the NB Dockerfile
+SRC_TAR=/tmp/$NB_USER.tar
```

Comment:
Why do we need to do this? Can't we just pip install the packages into a directory other than home? So that we don't have to reinstall them on startup?

Reply:
@jlewi I'm kind of struggling here. It seems like there is a minimal set of content from the Dockerfile build that should be put aside and then loaded into a "fresh" `/home/jovyan` including:
* gcloud config
* jupyter_notebook_config.py
* requirements.txt (as a user visibkle record of what was installed...sure they can do pip list, I suppose...)
/cc @ankushagarwal

Reply:
Deciding what we want and what we don't want to put on a PV is really tricky. I can argue that we might want to put python packages into PV as well since it's very unlikely that users will be using only the packages pre-installed in the image - users would want those packages to survive container restarts as well.

Putting things in PV is also really tricky because to put things on PV, we can't do it at build time and have to do it at runtime with solutions like copying the tarball. I don't have a good solution on how to do this cleanly.

Reply:
Nothing precludes a user from installing whatever they want into their home directory / PV after they've started the pod.

If at all possible I'd like to avoid having the Dockerfile put anything in /home/jovyan so we can avoid the complexity of having to copy files into /home after mounting the PVC.

gcloud config - I don't think we need this. My guess is it will be created automatically if a user runs gcloud.

requirements.txt I don't think a user visible record is that important. So we can just move it somewhere else or just forget about it for now.

.jupyter/jupyter_notebook_config.py  - Looks like this is autocreated when you start a notebook.

The only file I think we need is `.bashrc` and I think it would be fine to copy that on startup to the PVC (although even better would be if it uses the system default by default).

I ran a simple experiment and removed all the "." directories in my home directory and launched a notebook; it ran fine.

So I think if you've installed all the pip packages in a directory other than /home/jovyan I think we should be good to go.

Reply:
@jlewi, we seem to be fairly opinionated in the current Dockerfile....

gcloud:
```
gcloud config set core/disable_usage_reporting true && \
gcloud config set component_manager/disable_update_check true && \
gcloud config set metrics/environment github_docker_image
```

.jupyter/jupyter_notebook_config.py:
We definitely don't generate a default for this in the Dockerfile with `jupyter notebook --generate-config`. Instead we have [a custom one](https://github.com/kubeflow/kubeflow/blob/master/components/tensorflow-notebook-image/jupyter_notebook_config.py) copied in that seems specifically crafted.

requirements.txt:
Agreed.

Reply:
@willingc @yuvipanda @foxish Any reason why we would need a custom jupyternotebook_config?

For the gcloud config its fine to overwrite those defaults.

Reply:
> .jupyter/jupyter_notebook_config.py - Looks like this is autocreated when you start a notebook.

That's correct @jlewi. `jupyter_notebook_config.py` contains the default configuration for notebook when `jupyter notebook --generate-config` is run or the `jupyter notebook` is run for the first time. [Ref](https://jupyter-notebook.readthedocs.io/en/stable/config.html#config-file-and-command-line-options)

Reply:
OK, if there is no other content to preserve then I will update the PR.

Reply:
Thanks. I suggest we try it out and if we found we blew away something we shouldn't have we can add it back.